dask_planner/src/sql/logical.rs

```diff
@@ -140,19 +143,15 @@ impl PyLogicalPlan {
     }
 
     #[pyo3(name = "getRowType")]
-    pub fn row_type(&self) -> RelDataType {
-        let fields: &Vec<DFField> = self.original_plan.schema().fields();
-        let mut rel_fields: Vec<RelDataTypeField> = Vec::new();
-        for i in 0..fields.len() {
-            rel_fields.push(
-                RelDataTypeField::from(
-                    fields[i].clone(),
-                    self.original_plan.schema().as_ref().clone(),
-                )
-                .unwrap(),
-            );
-        }
-        RelDataType::new(false, rel_fields)
+    pub fn row_type(&self) -> PyResult<RelDataType> {
+        let schema = self.original_plan.schema();
+        let mut rel_fields: Vec<RelDataTypeField> = schema
+            .fields()
+            .iter()
+            .map(|f| RelDataTypeField::from(f, schema.as_ref()))
+            .collect::<Result<Vec<_>>>()
```

Comment:
The iterator is over items of type `Result` so `collect()` would normally return `Vec<Result<_>>` but we can use this turbofish notation to transform that into `Result<Vec<_>>` instead.

Reply:
much neater code here. Really like the turbofish use.